jupyterhub/apihandlers/users.py

```diff
@@ -259,11 +274,24 @@ async def patch(self, name):
         for key, value in data.items():
             if key == 'auth_state':
                 await user.save_auth_state(value)
+            elif key == 'groups':
+                groups = value
+                # add user to new groups
+                for name in groups:
+                    group = orm.Group.find(self.db, name=name)
+                    if group is None:
+                        raise web.HTTPError(409, "Group %s does not exist" % name)
+                    if group not in user.groups:
+                        user.groups.append(group)
+                # delete user from groups not in new groups
+                for group in user.groups:
+                    if group.name not in groups:
+                        user.groups.remove(group)
             else:
                 setattr(user, key, value)
         self.db.commit()
         user_ = self.user_model(user)
-        user_['auth_state'] = await user.get_auth_state()
+        user_['auth_state'] = await self._user_from_orm(user).get_auth_state()
```

Comment:
I see by this call to `_user_from_orm` that the above `find_user` really is required, so it should probably be put back.